core/templates/dev/head/pages/exploration-editor-page/improvements-tab/services/improvement-modal.service.ts

```diff
@@ -43,6 +43,129 @@ angular.module('oppia').factory('ImprovementModalService', [
       SuggestionModalForExplorationEditorService, ThreadDataService,
       ThreadStatusDisplayService, UrlInterpolationService, UserService) {
     return {
+      openPlaythroughModal: function(playthrough, index) {
+        $uibModal.open({
+          templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+            '/pages/exploration-editor-page/statistics-tab/templates/' +
+            'playthrough-modal.template.html'),
+          backdrop: true,
+          resolve: {
+            playthrough: function() {
+              return playthrough;
+            },
+            playthroughIndex: function() {
+              return index;
+            }
+          },
```

Comment:
resolve shouldn't be necessary here since you have access to the function arguments already.